In [11]:
import numpy as np
import pandas as pd

 Load a numeric column from a provided dataset into a NumPy array or Pandas Series:

In [32]:
df = pd.read_csv("data.csv")

df.head(10)

,Duration,Date,Pulse,Maxpulse,Calories
0,60,'2020/12/01',110,130,409.1
1,60,'2020/12/02',117,145,479.0
2,60,'2020/12/03',103,135,340.0
3,45,'2020/12/04',109,175,282.4
4,45,'2020/12/05',117,148,406.0
5,60,'2020/12/06',102,127,300.0
6,60,'2020/12/07',110,136,374.0
7,450,'2020/12/08',104,134,253.3
8,30,'2020/12/09',109,133,195.1
9,60,'2020/12/10',98,124,269.0


In [33]:
df.isnull().sum()

Duration    0
Date        1
Pulse       0
Maxpulse    0
Calories    2
dtype: int64

In [34]:
df["Calories"] = df["Calories"].fillna(df["Calories"].median())

In [35]:
df.drop_duplicates(inplace=True)

In [36]:
nums = np.array(df["Calories"])

In [37]:
print(nums)

[409.1 479.  340.  282.4 406.  300.  374.  253.3 195.1 269.  329.3 250.7
 345.3 379.3 275.  215.2 300.  291.2 323.  243.  364.2 282.  300.  246.
 334.5 250.  241.  291.2 280.  380.3 243. ]


Compute its mean, median, mode, standard deviation, and IQR:

In [29]:
mean = np.mean(nums)
median = np.median(nums)
mode = df["Calories"].mode()[0]

In [30]:
std = np.std(nums)
q1, q3 = np.percentile(nums, [25,75])
IQR = q3 - q1

In [31]:
print("Mean =", mean)
print("Median =", median)
print("mode =", mode)
print("std =", std)
print("IQR =", IQR)

Mean = 305.5516129032258
Median = 291.2
mode = 300.0
std = 63.185466228871284
IQR = 90.64999999999998


Identify whether the mean or median better represents a 'typical' value, and justify the choice: -->

The median (291.2) represents 'typical' value better than the mean (305.5516129032258), since if there an outlier the median ignores it, but the mean may calculate the outlier with the result and it will give us wrong answer.

Write a Markdown cell summarizing the column's center and spread in plain language: -->

if we take all central tendancy and compare them with std (63.185), we see that most values have moderate range around the avg., this means that the data nit too spread, but with some natural variation.